adxl345/adxl345.go

```diff
@@ -0,0 +1,252 @@
+	Dur              = 0x21 // Tap duration
+	Latent           = 0x22 // Tap latency
+	Window           = 0x23 // Tap window
+	ThreshAct        = 0x24 // Activity threshold
+	ThreshInact      = 0x25 // Inactivity threshold
+	TimeInact        = 0x26 // Inactivity time
+	ActInactCtl      = 0x27 // Axis control for activity/inactivity detection
+	ThreshFf         = 0x28 // Free-fall threshold
+	TapAxes          = 0x2A // Axis control for single tap/double tap
+	TapStatus        = 0x2B // Source of single tap/double tap
+	ActivityStatus   = 0x2A // Source of activity detection
+	InactivityStatus = 0x2B // Source of inactivity detection
+
+	// Control registers
+
+	BwRate     = 0x2C // Data rate and power mode control
+	PowerCtl   = 0x2D // Power saving features control
+	IntEnable  = 0x2E // Interrupt enable control
+	IntMap     = 0x2F // Interrupt mapping control
+	IntSource  = 0x30 // Source of interrupts
+	DataFormat = 0x31 // Data format control
+
+	// Data registers
+	DataX0 = 0x32 // X-Axis Data 0
+	DataX1 = 0x33 // X-Axis Data 1
+	DataY0 = 0x34 // Y-Axis Data 0
+	DataY1 = 0x35 // Y-Axis Data 1
+	DataZ0 = 0x36 // Z-Axis Data 0
+	DataZ1 = 0x37 // Z-Axis Data 1
+
+	// FIFO control
+	FifoCtl    = 0x38 // FIFO control
+	FifoStatus = 0x39 // FIFO status
+
+)
+
+// Sensitivity constants represents the sensitivity of the ADXL345.
+// The ADXL345 supports 4 sensitivity settings, ±2g, ±4g, ±8g, and ±16g, with the default being ±2g.
+// Sensitivity is an option that can be set at initialization in Opts.
+// You can set the sensitivity of the ADXL345 with the DataFormat register.
+const (
+	S2G  Sensitivity = 0x00 // Sensitivity at 2g
+	S4G  Sensitivity = 0x01 // Sensitivity at 4g
+	S8G  Sensitivity = 0x02 // Sensitivity at 8g
+	S16G Sensitivity = 0x03 // Sensitivity at 16g
+)
+
+var (
+	SpiFrequency = physic.MegaHertz * 2
+	SpiMode      = spi.Mode3 // Defines the base clock signal, along with the polarity and phase of the data signal.
+	SpiBits      = 8
+)
+
+var DefaultOpts = Opts{
+	ExpectedDeviceID: 0xE5,
+	Sensitivity:      S2G,
+}
+
+type Opts struct {
+	ExpectedDeviceID byte        // Expected device ID used to verify that the device is an ADXL345.
+	Sensitivity      Sensitivity // Sensitivity of the device (2G, 4G, 8G, 16G)
+}
+
+// Dev is a driver for the ADXL345 accelerometer
+// It uses the SPI interface to communicate with the device.
+type Dev struct {
+	c    mmr.Dev16
+	name string
+	s    spi.Conn
+	// The sensitivity of the device (2G, 4G, 8G, 16G)
+	// Set to 2G by default, can be changed in the Opts at initialization.
+	sensitivity Sensitivity
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("ADXL345{Sensitivity:%s}", d.sensitivity)
+}
+
+// New creates a new ADXL345 Dev or returns an error.
+// The bus and chip parameters define the SPI bus and chip select to use.
+// The SPI s is configured.
+// The device is turned on.
+// The device is verified to be an ADXL345.
+func New(p spi.Port, o *Opts) (*Dev, error) {
```

Comment:
I'd recommend to name it NewSPI right away so if I2C is added later, it's a non-breaking change.

Reply:
Done